include/circt/Dialect/HW/HWStructure.td

```diff
@@ -458,3 +458,20 @@ def GlobalRefOp : HWOp<"globalRef", [IsolatedFromAbove, Symbol]> {
 
   let verifier = "return this->verifyGlobalRef();";
 }
+
+def ProbeOp : HWOp<"probe", []> {
+  let summary = "Probe values for use in remote references";
+  let description = [{
+    Captures values without binding to any accidental name.  This allows
+    capturing names holding values of interest while allowing the name to
+    resolved only at emission time.
+  }];
+
+  let arguments = (ins SymbolNameAttr:$inner_sym,
+                       Variadic<AnyType>:$operands);
```

Comment:
Why multiple values rather than just one? The semantics seem clearer with only one.

Reply:
It would increase complexity and representation size drastically.  Instead of a bind being 2 ops (probe and bind) with the arguments listed once and 1 total added symbols, it would have one symbol and one op per argument, significantly bloating the IR for what is, essentially just a dataflow sink which isn't ever transformed.